webapp/sources/rudder/rudder-web/src/main/scala/com/normation/plugins/PublicPlugin.scala

```diff
@@ -94,6 +94,28 @@ trait DefaultPluginDef extends RudderPluginDef {
       PluginVersion(0,0,1, s"ERROR-PARSING-VERSION: ${versionString}")
     )
   }
+  override lazy val versionInfo = {
+    val info = buildConf.getString("version-info")
+    if (info.isEmpty) None else Some(info)
+
+//    val content = for {
+//      info <- IO.effect {buildConf.getString("version-info")}
+//    } yield {
+//      Some(info)
+//    }
+//    content.catchSome {
+//      case missingErr: ConfigException.Missing =>
+//        ApplicationLogger.info("Missing version-info in build.conf")
+//        None.succeed
+//      case typeErr: ConfigException.WrongType =>
+//        ApplicationLogger.error("Wrong type version-info in build.conf")
+//        None.succeed
+//      case e =>
+//        ApplicationLogger.error(s"Error occurs when trying to get version-info in build.conf, cause by : ${e}")
+//        None.succeed
+//    }
```

Comment:
My Intention was to try to get the `version-info` parameter in build.conf, as this parameter is not mandatory, `getString` can fail and throw exception. In this scenario I think we don't want the program fail because of the missing parameter, so I try to catch the exceptions and log the error and return `None`
If the `getString` succeed (no Throwable) I would like to get an `Some(String)`, Nose otherwise bu,t I don't think it's the right thing to do in ZIO

Does `versionInfo` have to be an :
 + `Option[String]` -> match/case at the end (when we want to display in the webapp)
+ `IO[Throwable, Option[String]]`

Reply:
just a ```try { val x = buildConf.getString("version-info"); if(x.isEmpty) None else Some(x) } catch { ex:ConfigException => None }```

(I'm not sure about exception name, but you have several example for rudder properties in `RudderConfig` file)

Reply:
You can catch any exception also, or use net.liftweb.util.tryo that returns a box and then transform into an option